wiring/src/spark_wiring_json.cpp

```diff
@@ -542,7 +543,7 @@ void spark::JSONWriter::writeEscaped(const char *str, size_t size) {
     const char *s = str;
     while (s != end) {
         const char c = *s;
-        if (c == '"' || c == '\\' || (c >= 0 && c <= 0x1f)) {
+        if (c == '"' || c == '\\' || std::isprint(c) || std::iscntrl(c)) {
```

Comment:
`std::isprint(c)` here will cause all printable characters to be escaped. Also, I believe you should explicitly cast the argument to `unsigned char` for portability when using functions from ctype.h.

Reply:
I think I misread that as `0x7f` which would be matching `isprint()` + `iscntrl()`. :+1:

Reply:
Fixed